### PR TITLE
Wheels: support ocl binary distributions

### PR DESCRIPTION
**Pull request by isuruf**
Scripts for building pocl and oclgrind are at https://github.com/isuruf/ocl-wheels.

cc @gmagno

### CODE REVIEW

**Comment by isuruf** [comment]
still a WIP. Pocl doesn't work properly

**Comment by isuruf** [comment]
Ready for a review now. There's one catch. `pyopencl` and `pocl-binary-distribution` have to be installed side-by-side. Otherwise `pyopencl` will not see `pocl`

**Comment by isuruf** [comment]
Ping. Added suport for `pyopencl[oclgrind]` as well.

File: travis/build-wheels.sh [hunk 1]
```diff
@@ -51,6 +53,10 @@ done
 /opt/python/cp37-cp37m/bin/pip install delocate
 /opt/python/cp37-cp37m/bin/python /io/travis/fix-wheel.py /deps/ocl-icd/COPYING
 
+if [[ "${TWINE_USERNAME}" == "" ]]; then
+    exit 0
```
Comment (inducer) [inline travis/build-wheels.sh:58]: Could you add a message here, to avoid confusion on why there's no upload?

**Review by inducer — CHANGES_REQUESTED** [review]
Could you also hack the docs (`docs/misc.rst`) to highlight this as an alternate install option?

**Review by inducer — APPROVED** [review]

**Comment by inducer** [comment]
Thanks @isuruf!

**Comment by inducer** [comment]
Releasing 2018.2.5 mainly as an excuse to get these wheels out there.

**Comment by isuruf** [comment]
Looks like it worked. `pip install "pyopencl[pocl, oclgrind]"` gave me 2 working opencl implementations.

**Comment by inducer** [comment]
There are wheels! https://pypi.org/project/pyopencl/2018.2.5/#files

@gmagno @isuruf Could you check that they work?

And the docs got updated: https://documen.tician.de/pyopencl/misc.html#installing-from-pypi-with-linux-wheels

I wonder whether we should promote wheels first--i.e. move that above the conda-based method.

**Comment by inducer** [comment]
@isuruf (a) you beat me to it! :) :+1: 
(b) This is so cool. Thanks for working on this.

**Comment by gmagno** [comment]
oh man, this is great! :rocket: 
It looks good on my system.

image

**Pull request merged**